core/templates/domain/feedback_message/ThreadMessageObjectFactory.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of thread message
+ * domain objects.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { ThreadMessageSummary, ThreadMessageSummaryObjectFactory } from
+  'domain/feedback_message/ThreadMessageSummaryObjectFactory';
+
+export class ThreadMessage {
+  authorUsername: string;
+  createdOn: number;
+  entityType: string;
+  entityId: string;
+  messageId: number;
+  receivedViaEmail: boolean;
+  text: string = '';
+  updatedStatus: string = null;
+  updatedSubject: string = null;
+  summary: ThreadMessageSummary;
+
+  constructor(
+      authorUsername: string, createdOn: number, entityType: string,
+      entityId: string, messageId: number, receivedViaEmail: boolean,
+      text: string, updatedStatus: string, updatedSubject: string,
+      summary: ThreadMessageSummary) {
+    this.authorUsername = authorUsername;
+    this.createdOn = createdOn;
+    this.entityType = entityType;
+    this.entityId = entityId;
+    this.messageId = messageId;
+    this.receivedViaEmail = receivedViaEmail;
+    this.text = text;
+    this.updatedStatus = updatedStatus;
+    this.updatedSubject = updatedSubject;
+    this.summary = summary;
+  }
+
+  isNonempty(): boolean {
```

Comment:
Just a check: here and elsewhere, would hasText() or hasNonemptyText() work better? Seems closer to what this method is evaluating.